From e4c509968deac66f7ac1086f862d7780ea4e611e Mon Sep 17 00:00:00 2001
From: Andy Chung <Andy.Chung@amd.com>
Date: Mon, 25 May 2026 14:05:38 +0800
Subject: [PATCH] configuration: anacapa: Update Mortaro sensors under NIC
 inventory

Summary:
1. Migrate Mortaro NIC sensors to the NIC inventory.
2. Add additional PLDM sensor support.
3. Update Thor3 NIC thresholds and sensor name to align with Mortaro.
---
 .../amd/vulcano_1O800_800g_ocp.json           | 142 +++++++++++++++++-
 configurations/broadcomm/800g_1p_ocp.json     |  21 ++-
 2 files changed, 155 insertions(+), 8 deletions(-)

diff --git a/configurations/amd/vulcano_1O800_800g_ocp.json b/configurations/amd/vulcano_1O800_800g_ocp.json
index 84ecebc..60996a4 100644
--- a/configurations/amd/vulcano_1O800_800g_ocp.json
+++ b/configurations/amd/vulcano_1O800_800g_ocp.json
@@ -5,9 +5,149 @@
             "Bus": "$bus",
             "Name": "AMD OCP NIC FRU",
             "Type": "EEPROM"
+        },
+        {
+            "EntityId": 11,
+            "EntityInstance": 0,
+            "MaxValue": 255,
+            "MinValue": -255,
+            "Name": "NIC$bus - 39 TEMP_C",
+            "Timeout": 30.0,
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Name": "upper non critical",
+                    "Severity": 0,
+                    "Value": 95
+                },
+                {
+                    "Direction": "greater than",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 100
+                },
+                {
+                    "Direction": "greater than",
+                    "Name": "upper non recoverable",
+                    "Severity": 4,
+                    "Value": 105
+                }
+            ],
+            "Type": "ExternalSensor",
+            "Units": "DegreesC"
+        },
+        {
+            "EntityId": 11,
+            "EntityInstance": 1,
+            "MaxValue": 255,
+            "MinValue": -255,
+            "Name": "NIC$bus - 39 VR_TEMP_C",
+            "Timeout": 30.0,
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Name": "upper non critical",
+                    "Severity": 0,
+                    "Value": 95
+                },
+                {
+                    "Direction": "greater than",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 100
+                },
+                {
+                    "Direction": "greater than",
+                    "Name": "upper non recoverable",
+                    "Severity": 4,
+                    "Value": 105
+                }
+            ],
+            "Type": "ExternalSensor",
+            "Units": "DegreesC"
+        },
+        {
+            "EntityId": 11,
+            "EntityInstance": 2,
+            "MaxValue": 255,
+            "MinValue": -255,
+            "Name": "NIC$bus - 39 BOARD_TEMP_C",
+            "Timeout": 30.0,
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Name": "upper non critical",
+                    "Severity": 0,
+                    "Value": 75
+                },
+                {
+                    "Direction": "greater than",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 80
+                },
+                {
+                    "Direction": "greater than",
+                    "Name": "upper non recoverable",
+                    "Severity": 4,
+                    "Value": 85
+                }
+            ],
+            "Type": "ExternalSensor",
+            "Units": "DegreesC"
+        },
+        {
+            "EntityId": 11,
+            "EntityInstance": 3,
+            "MaxValue": 255,
+            "MinValue": -255,
+            "Name": "NIC$bus - 39 OSFP_TEMP_C",
+            "Timeout": 30.0,
+            "Type": "ExternalSensor",
+            "Units": "DegreesC"
+        },
+        {
+            "EntityId": 11,
+            "EntityInstance": 4,
+            "MaxValue": 3000,
+            "MinValue": 0,
+            "Name": "NIC$bus - 39 VR_PWR_W",
+            "Timeout": 30.0,
+            "Type": "ExternalSensor",
+            "Units": "Watts"
+        },
+        {
+            "EntityId": 11,
+            "EntityInstance": 5,
+            "MaxValue": 3000,
+            "MinValue": 0,
+            "Name": "NIC$bus - 39 ASIC_PWR_W",
+            "Timeout": 30.0,
+            "Type": "ExternalSensor",
+            "Units": "Watts"
+        },
+        {
+            "EntityId": 11,
+            "EntityInstance": 6,
+            "MaxValue": 255,
+            "MinValue": 0,
+            "Name": "NIC$bus - 39 VR_VIN_VOLT_V",
+            "Timeout": 30.0,
+            "Type": "ExternalSensor",
+            "Units": "Volts"
+        },
+        {
+            "EntityId": 11,
+            "EntityInstance": 7,
+            "MaxValue": 255,
+            "MinValue": 0,
+            "Name": "NIC$bus - 39 ASIC_VOLT_V",
+            "Timeout": 30.0,
+            "Type": "ExternalSensor",
+            "Units": "Volts"
         }
     ],
-    "Name": "AMD OCP NIC $bus",
+    "Name": "OCP NIC $bus - 39",
     "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'VULCANO-1O800-OCP*', 'BOARD_MANUFACTURER': 'AMD'})",
     "Type": "Board",
     "xyz.openbmc_project.Inventory.Decorator.Asset": {
diff --git a/configurations/broadcomm/800g_1p_ocp.json b/configurations/broadcomm/800g_1p_ocp.json
index c80209c..67d65f1 100644
--- a/configurations/broadcomm/800g_1p_ocp.json
+++ b/configurations/broadcomm/800g_1p_ocp.json
@@ -12,28 +12,35 @@
             "Labels": [
                 "temp2"
             ],
-            "Name": "BRB_L_FE_NIC0_TEMP_C",
-            "Name1": "BRB_L_FE_NIC0_TEMP_C",
+            "Name": "NIC$bus - 39 TEMP_C",
+            "Name1": "NIC$bus - 39 TEMP_C",
             "Thresholds": [
                 {
-                    "Direction": "less than",
+                    "Direction": "greater than",
                     "Index": 2,
-                    "Name": "lower critical",
-                    "Severity": 1,
-                    "Value": 5
+                    "Name": "upper non critical",
+                    "Severity": 0,
+                    "Value": 95
                 },
                 {
                     "Direction": "greater than",
                     "Index": 2,
                     "Name": "upper critical",
                     "Severity": 1,
+                    "Value": 100
+                },
+                {
+                    "Direction": "greater than",
+                    "Index": 2,
+                    "Name": "upper non recoverable",
+                    "Severity": 4,
                     "Value": 105
                 }
             ],
             "Type": "TMP421"
         }
     ],
-    "Name": "BRCM OCP NIC",
+    "Name": "OCP NIC $bus - 39",
     "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Broadcom BCM957708 1x800G OCP', 'BOARD_MANUFACTURER': 'Broadcom'})",
     "Type": "Board",
     "xyz.openbmc_project.Inventory.Decorator.Asset": {
-- 
2.53.0-Meta

